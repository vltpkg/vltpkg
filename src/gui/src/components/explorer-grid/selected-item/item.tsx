import { Outlet } from 'react-router'
import { Card } from '@/components/ui/card.tsx'
import { Tabs, TabsList } from '@/components/ui/tabs.tsx'
import {
  SelectedItemProvider,
  useSelectedItemStore,
} from '@/components/explorer-grid/selected-item/context.tsx'
import { InsightTabButton } from '@/components/explorer-grid/selected-item/tabs-insight.tsx'
import { OverviewTabButton } from '@/components/explorer-grid/selected-item/tabs-overview.tsx'
import { VersionsTabButton } from '@/components/explorer-grid/selected-item/tabs-versions.tsx'
import { TabsManifestButton } from '@/components/explorer-grid/selected-item/tabs-manifest.tsx'
import { DependenciesTabsButton } from '@/components/explorer-grid/selected-item/tabs-dependencies/index.tsx'
import { ItemHeader } from '@/components/explorer-grid/selected-item/item-header.tsx'
import type { Tab } from '@/components/explorer-grid/selected-item/context.tsx'
import type { GridItemData } from '@/components/explorer-grid/types.ts'

export const Item = ({ item }: { item: GridItemData }) => {
  return (
    <SelectedItemProvider selectedItem={item}>
      <div className="relative">
        <Card className="relative border-muted">
          <ItemHeader />
          <SelectedItemTabs />
        </Card>
      </div>
    </SelectedItemProvider>
  )
}

const SelectedItemTabs = () => {
  const selectedItem = useSelectedItemStore(
    state => state.selectedItem,
  )
  const activeTab = useSelectedItemStore(state => state.activeTab)
  const setActiveTab = useSelectedItemStore(
    state => state.setActiveTab,
  )

  const handleTabChange = (newTab: string) =>
    setActiveTab(newTab as Tab)

  return (
    <div className="w-full">
      <Tabs
        uniqueId={selectedItem.id}
        onValueChange={handleTabChange}
        value={activeTab}>
        <TabsList variant="ghost" className="w-full gap-2 px-6">
          <OverviewTabButton />
          <TabsManifestButton />
          <InsightTabButton />
          <VersionsTabButton />
          <DependenciesTabsButton />
        </TabsList>
        <Outlet />
      </Tabs>
    </div>
  )
}
