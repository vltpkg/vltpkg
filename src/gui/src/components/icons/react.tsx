export const React = (props: React.SVGProps<SVGSVGElement>) => {
  return (
    <svg
      width="500"
      height="500"
      viewBox="0 0 500 500"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}>
      <g clip-path="url(#clip0_2967_809)">
        <path
          d="m250.504 284.862c270.046 284.862 285.889 269.029 285.889 249.499c285.889 229.968 270.046 214.136 250.504 214.136c230.961 214.136 215.119 229.968 215.119 249.499c215.119 269.029 230.961 284.862 250.504 284.862z"
          fill="currentColor"
        />
        <path
          d="m250.501 321.96c355.363 321.96 440.371 289.522 440.371 249.509c440.371 209.495 355.363 177.058 250.501 177.058c145.639 177.058 60.6318 209.495 60.6318 249.509c60.6318 289.522 145.639 321.96 250.501 321.96z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.724 285.725c240.155 376.482 310.768 433.837 345.442 413.83c380.116 393.823 365.721 304.031 313.29 213.274c260.859 122.517 190.247 65.1621 155.572 85.1689c120.898 105.176 135.293 194.968 187.724 285.725z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.726 213.277c135.295 304.034 120.9 393.826 155.574 413.833c190.248 433.84 260.861 376.485 313.292 285.728c365.723 194.971 380.118 105.179 345.443 85.172c310.769 65.1652 240.157 122.52 187.726 213.277z"
          stroke="currentColor"
          stroke-width="18.913"
        />
      </g>
      <defs>
        <clipPath id="clip0_2967_809">
          <rect
            width="397"
            height="353"
            fill="currentColor"
            transform="translate(52 73)"
          />
        </clipPath>
      </defs>
    </svg>
  )
}
