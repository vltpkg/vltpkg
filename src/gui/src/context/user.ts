import { createContext } from 'react-router'

import type { User } from '@/state/auth.ts'

/**
 * Router context for authenticated user
 * Used by auth middleware to pass user data to loaders/components
 */
export const userContext = createContext<User | null>(null)
