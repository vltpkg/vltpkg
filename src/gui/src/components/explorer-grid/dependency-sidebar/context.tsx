import { createContext, useEffect, useContext, useRef } from 'react'
import { createStore, useStore } from 'zustand'
import { useToast } from '@/components/hooks/use-toast.ts'
import { useNavigate } from 'react-router'
import { useGraphStore } from '@/state/index.ts'
import { addRemoveDependency } from '@/lib/add-remove-dependency.ts'
import { deepEqual } from '@/utils/deep-equal.ts'

import type { StoreApi } from 'zustand'
import type { PropsWithChildren } from 'react'
import type { GridItemData } from '@/components/explorer-grid/types.ts'
import type { DependencySideBarProps } from '@/components/explorer-grid/dependency-sidebar/index.tsx'
import type { Operation } from '@/lib/add-remove-dependency.ts'
import type { Filter } from './filter-config.ts'

type DependencySidebarState = {
  importerId?: DependencySideBarProps['importerId']
  dependencies: DependencySideBarProps['dependencies']
  filteredDependencies: DependencySideBarProps['dependencies']
  filters: Filter[]
  searchTerm: string
  addedDependencies: string[]
  uninstalledDependencies: DependencySideBarProps['uninstalledDependencies']
  inProgress: boolean
  error?: string
  dependencyPopoverOpen?: boolean
}
type DependencySidebarAction = {
  setAddedDependencies: (
    deps: DependencySidebarState['addedDependencies'],
  ) => void
  setFilteredDependencies: (
    deps: DependencySidebarState['filteredDependencies'],
  ) => void
  setFilters: (filters: DependencySidebarState['filters']) => void
  setSearchTerm: (
    search: DependencySidebarState['searchTerm'],
  ) => void
  setInProgress: (bool: DependencySidebarState['inProgress']) => void
  onDependencyClick: (item: GridItemData) => () => undefined
  setError: (str: DependencySidebarState['error']) => void
  setDependencyPopoverOpen: (
    o: DependencySidebarState['dependencyPopoverOpen'],
  ) => void
}

export type DependencySidebarStore = DependencySidebarState &
  DependencySidebarAction

const DependencySidebarContext = createContext<
  StoreApi<DependencySidebarStore> | undefined
>(undefined)

type DependencySidebarProviderProps = PropsWithChildren & {
  dependencies: DependencySidebarState['dependencies']
  importerId?: DependencySidebarState['importerId']
  onDependencyClick: DependencySidebarAction['onDependencyClick']
  uninstalledDependencies: DependencySidebarState['uninstalledDependencies']
}

/**
 * Deeply compares incoming props and updates the store only if values have changed.
 * Prevents unnecessary store updates and downstream rerenders.
 */
const useSyncStoreProps = <T,>(
  store: StoreApi<T>,
  props: Partial<T>,
): void => {
  const prevPropsRef = useRef<Partial<T>>({})

  useEffect(() => {
    const prev = prevPropsRef.current
    const changedKeys = Object.keys(props).filter(key => {
      const typedKey = key as keyof T
      return !deepEqual(props[typedKey], prev[typedKey])
    })

    if (changedKeys.length > 0) {
      const updated: Partial<T> = {}
      changedKeys.forEach(key => {
        const typedKey = key as keyof T
        updated[typedKey] = props[typedKey]
      })
      store.setState(updated)
      prevPropsRef.current = props
    }
  }, [props, store])
}

export const DependencySidebarProvider = ({
  dependencies,
  uninstalledDependencies,
  importerId,
  onDependencyClick,
  children,
}: DependencySidebarProviderProps) => {
  const query = useGraphStore(state => state.query)
  const dependencySidebarStore = useRef(
    createStore<DependencySidebarStore>(set => ({
      dependencies,
      filteredDependencies: dependencies,
      uninstalledDependencies,
      importerId,
      filters: [],
      searchTerm: '',
      onDependencyClick,
      addedDependencies: [],
      inProgress: false,
      error: undefined,
      dependencyPopoverOpen: false,
      setAddedDependencies: (
        deps: DependencySidebarState['addedDependencies'],
      ) => set(() => ({ addedDependencies: deps })),
      setFilters: (filters: DependencySidebarState['filters']) =>
        set(() => ({ filters })),
      setSearchTerm: (search: string) =>
        set(() => ({ searchTerm: search })),
      setFilteredDependencies: (
        filteredDependencies: DependencySidebarState['filteredDependencies'],
      ) => set(() => ({ filteredDependencies })),
      setInProgress: (bool: DependencySidebarState['inProgress']) =>
        set(() => ({ inProgress: bool })),
      setError: (str: DependencySidebarState['error']) =>
        set(() => ({ error: str })),
      setDependencyPopoverOpen: (
        o: DependencySidebarState['dependencyPopoverOpen'],
      ) => set(() => ({ dependencyPopoverOpen: o })),
    })),
  ).current

  /** The `filteredDependencies` by default will the dependencies themselves */
  useEffect(() => {
    dependencySidebarStore.setState(() => ({
      filteredDependencies: dependencies,
    }))
  }, [dependencies, dependencySidebarStore])

  /** Reset the filters when navigating */
  useEffect(() => {
    dependencySidebarStore.setState(() => ({
      filters: [],
    }))
  }, [query, dependencySidebarStore])

  useSyncStoreProps(dependencySidebarStore, {
    dependencies,
    uninstalledDependencies,
    importerId,
    onDependencyClick,
  })

  return (
    <DependencySidebarContext.Provider value={dependencySidebarStore}>
      {children}
    </DependencySidebarContext.Provider>
  )
}

export const useDependencySidebarStore = <T,>(
  selector: (state: DependencySidebarStore) => T,
) => {
  const store = useContext(DependencySidebarContext)
  if (!store) {
    throw new Error(
      'useDependencySidebarStore must be used within a DependencySidebarProvider',
    )
  }
  return useStore(store, selector)
}

/**
 * usePopover is scoped to the DependencySidebarProvider, and is intended
 * for controlling the state of the `addedDependenciesPopover` component.
 *
 * It will throw an error if used outside of the DependencySidebarProvider.
 */
export const usePopover = () => {
  const setInProgress = useDependencySidebarStore(
    state => state.setInProgress,
  )
  const error = useDependencySidebarStore(state => state.error)
  const setError = useDependencySidebarStore(state => state.setError)

  const dependencyPopoverOpen = useDependencySidebarStore(
    state => state.dependencyPopoverOpen,
  )
  const setDependencyPopoverOpen = useDependencySidebarStore(
    state => state.setDependencyPopoverOpen,
  )

  const toggleAddDepPopover = () => {
    if (dependencyPopoverOpen && error) {
      setError('')
      setInProgress(false)
    }
    setDependencyPopoverOpen(!dependencyPopoverOpen)
  }

  return {
    toggleAddDepPopover,
    dependencyPopoverOpen,
    setDependencyPopoverOpen,
  }
}

export const useOperation = () => {
  const navigate = useNavigate()
  const { toast } = useToast()
  const { toggleAddDepPopover } = usePopover()

  const updateStamp = useGraphStore(state => state.updateStamp)
  const updateErrorCause = useGraphStore(
    state => state.updateErrorCause,
  )

  const importerId = useDependencySidebarStore(
    state => state.importerId,
  )
  const setError = useDependencySidebarStore(state => state.setError)
  const setInProgress = useDependencySidebarStore(
    state => state.setInProgress,
  )
  const addedDependencies = useDependencySidebarStore(
    state => state.addedDependencies,
  )
  const setAddedDependencies = useDependencySidebarStore(
    state => state.setAddedDependencies,
  )

  const operation = async ({
    item,
    operationType,
  }: {
    item: {
      name: GridItemData['name']
      version: GridItemData['version']
      type?: string
    }
    operationType: Operation
  }) => {
    if (!importerId) return

    const onSuccessful =
      operationType === 'install' ?
        () => {
          toggleAddDepPopover()
          setAddedDependencies([item.name, ...addedDependencies])
        }
      : () => {
          const packageName = item.name
          setAddedDependencies(
            addedDependencies.filter(item => item !== packageName),
          )
        }

    try {
      await addRemoveDependency({
        operation: operationType,
        setError,
        setInProgress,
        updateStamp,
        toast,
        importerId,
        name: item.name,
        version: item.version,
        type: item.type,
        onSuccessful,
      })
    } catch (err) {
      console.error(err)
      void navigate('/error')
      updateErrorCause(
        'Unexpected error trying to install dependency.',
      )
    }
  }

  return { operation }
}
