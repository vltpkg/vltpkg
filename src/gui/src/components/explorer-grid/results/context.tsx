import { useEffect, useRef, useContext, createContext } from 'react'
import { useStore, createStore } from 'zustand'
import {
  useNavigate,
  useSearchParams,
  useLocation,
} from 'react-router'
import { PAGE_SIZE_OPTIONS } from '@/components/explorer-grid/results/page-options.tsx'

import type { PropsWithChildren } from 'react'
import type { StoreApi } from 'zustand'
import type { PageSizeOption } from '@/components/explorer-grid/results/page-options.tsx'
import type { GridItemData } from '@/components/explorer-grid/types.tsx'

export type ResultsSortBy =
  | 'alphabetical'
  | 'version'
  | 'dependencyType'
  | 'dependents'
  | 'moduleType'
  | 'overallScore'

type ResultsStoreState = {
  /**
   * The current page number
   */
  page: number
  /**
   * The total number of pages
   */
  totalPages: number
  /**
   * The count of items on each page
   */
  pageSize: PageSizeOption
  /**
   * The items to display on the current page
   */
  pageItems: GridItemData[]
  /**
   * A collection of all the items available
   */
  allItems: GridItemData[]
  /**
   * Current sort key for list view
   */
  sortBy: ResultsSortBy
}

type ResultsStoreAction = {
  /**
   * Sets the current page
   */
  setPage: (page: ResultsStoreState['page']) => void
  /**
   * Sets the count of items on each page
   */
  setPageSize: (pageSize: ResultsStoreState['pageSize']) => void
  /**
   * Sets the sort key for the results list
   */
  setSortBy: (sortBy: ResultsSortBy) => void
}

export type ResultsStore = ResultsStoreState & ResultsStoreAction

interface ResultsProviderProps extends PropsWithChildren {
  allItems: ResultsStoreState['allItems']
}

const ResultsContext = createContext<
  StoreApi<ResultsStore> | undefined
>(undefined)

export const ResultsProvider = ({
  children,
  allItems,
}: ResultsProviderProps) => {
  const navigate = useNavigate()
  const [searchParams, setSearchParams] = useSearchParams()
  const { pathname } = useLocation()

  const latestSearchParamsRef = useRef(searchParams)
  useEffect(() => {
    latestSearchParamsRef.current = searchParams
  }, [searchParams])

  const resultsStore = useRef(
    createStore<ResultsStore>((_, get) => ({
      page: Number(searchParams.get('page') ?? '1'),
      totalPages: 1,
      pageSize: Number(
        searchParams.get('pageSize') ??
          PAGE_SIZE_OPTIONS[0].toString(),
      ) as PageSizeOption,
      pageItems: [],
      allItems,
      sortBy: (() => {
        const raw = (searchParams.get('sort') ??
          'alphabetical') as ResultsSortBy
        const valid: ResultsSortBy[] = [
          'alphabetical',
          'version',
          'dependencyType',
          'dependents',
          'moduleType',
          'overallScore',
        ]
        return valid.includes(raw) ? raw : 'alphabetical'
      })(),
      setPage: (page: ResultsStoreState['page']) => {
        const next = new URLSearchParams(
          latestSearchParamsRef.current,
        )
        next.set('page', String(page))
        // keep current pageSize; if missing, use the store's current value
        if (!next.get('pageSize')) {
          next.set('pageSize', String(get().pageSize))
        }
        setSearchParams(next, { replace: true })
      },
      setPageSize: (pageSize: ResultsStoreState['pageSize']) => {
        const next = new URLSearchParams(
          latestSearchParamsRef.current,
        )
        next.set('pageSize', String(pageSize))
        // reset page to 1 when page size changes
        next.set('page', '1')
        setSearchParams(next, { replace: true })
      },
      setSortBy: (sortBy: ResultsSortBy) => {
        const next = new URLSearchParams(
          latestSearchParamsRef.current,
        )
        next.set('sort', sortBy)
        setSearchParams(next, { replace: true })
      },
    })),
  ).current

  /**
   * When new `items` are available: we assume there are results,
   * and we navigate to `.../<encodedUrlQuery>?page=1`
   */
  useEffect(() => {
    // only run this logic when we actually have a result set that implies “list view”
    // i.e., more than one item. Skip while loading (length === 0) and skip the single-item case.
    if (allItems.length > 1) {
      // remove trailing "/overview" (with or without trailing slash)
      const newPath = pathname.replace(/\/overview\/?$/, '')

      // only navigate if path is actually changing
      if (newPath !== pathname) {
        void navigate(newPath, { replace: true }) // avoid history spam
      }

      // ensure page=1 is present; clone instead of mutating in place
      if (!searchParams.get('page')) {
        const next = new URLSearchParams(searchParams)
        next.set('page', '1')
        next.set('pageSize', PAGE_SIZE_OPTIONS[0].toString())
        setSearchParams(next, { replace: true })
      }
    }
  }, [allItems, pathname, navigate, searchParams, setSearchParams])

  // keep store in sync with URL and data; clamp and validate values
  useEffect(() => {
    const defaultPageSize = PAGE_SIZE_OPTIONS[0]
    const rawPageSize = Number(
      searchParams.get('pageSize') ?? String(defaultPageSize),
    )
    const validPageSize =
      (
        (PAGE_SIZE_OPTIONS as unknown as number[]).includes(
          rawPageSize,
        )
      ) ?
        rawPageSize
      : defaultPageSize

    const totalPages = Math.max(
      1,
      Math.ceil(allItems.length / Math.max(1, validPageSize)),
    )

    const rawPage = Number(searchParams.get('page') ?? '1')
    const clampedPage = Math.min(Math.max(rawPage, 1), totalPages)

    const rawSort = (searchParams.get('sort') ??
      'alphabetical') as ResultsSortBy
    const validSorts: ResultsSortBy[] = [
      'alphabetical',
      'version',
      'dependencyType',
      'dependents',
      'moduleType',
      'overallScore',
    ]
    const sortBy =
      validSorts.includes(rawSort) ? rawSort : 'alphabetical'

    const start = (clampedPage - 1) * validPageSize
    const pageSlice = allItems.slice(start, start + validPageSize)

    const compareString = (x?: string, y?: string) =>
      (x ?? '').localeCompare(y ?? '', undefined, {
        numeric: true,
        sensitivity: 'base',
      })
    const compareNumber = (
      x?: number | null,
      y?: number | null,
      desc = false,
    ) => {
      const nx = x ?? -Infinity
      const ny = y ?? -Infinity
      return desc ? ny - nx : nx - ny
    }

    const pageItems = [...pageSlice].sort((a, b) => {
      switch (sortBy) {
        case 'alphabetical':
          return compareString(a.name, b.name)
        case 'version': {
          const va = a.version
          const vb = b.version
          return compareString(va, vb)
        }
        case 'dependencyType':
          return compareString(a.type, b.type)
        case 'dependents':
          return compareNumber(a.size, b.size, true)
        case 'moduleType': {
          const ma = a.to?.manifest?.type ?? ''
          const mb = b.to?.manifest?.type ?? ''
          return compareString(ma, mb)
        }
        case 'overallScore': {
          const scoreA = a.to?.insights.score
          const scoreB = b.to?.insights.score
          const sa = scoreA ? scoreA.overall : null
          const sb = scoreB ? scoreB.overall : null
          return compareNumber(sa, sb, true)
        }
      }
    })

    // reflect clamped/validated values in URL if needed
    if (
      rawPage !== clampedPage ||
      rawPageSize !== validPageSize ||
      rawSort !== sortBy
    ) {
      const next = new URLSearchParams(searchParams)
      next.set('page', String(clampedPage))
      next.set('pageSize', String(validPageSize))
      next.set('sort', sortBy)
      setSearchParams(next, { replace: true })
    }

    resultsStore.setState({
      page: clampedPage,
      pageSize: validPageSize as PageSizeOption,
      totalPages,
      pageItems,
      allItems,
      sortBy,
    })
  }, [allItems, searchParams, resultsStore, setSearchParams])

  return (
    <ResultsContext.Provider value={resultsStore}>
      {children}
    </ResultsContext.Provider>
  )
}

export const useResultsStore = <T,>(
  selector: (state: ResultsStore) => T,
) => {
  const store = useContext(ResultsContext)
  if (!store) {
    throw new Error(
      'useResultsStore must be used within a ResultsProvider',
    )
  }
  return useStore(store, selector)
}
