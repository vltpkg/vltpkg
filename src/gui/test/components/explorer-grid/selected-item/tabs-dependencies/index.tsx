import { test, expect, vi, afterEach } from 'vitest'
import { cleanup, render } from '@testing-library/react'
import html from 'diffable-html'
import { useGraphStore as useStore } from '@/state/index.ts'
import { useSelectedItemStore } from '@/components/explorer-grid/selected-item/context.tsx'
import { SELECTED_ITEM } from '../__fixtures__/item.ts'
import {
  DependenciesTabsButton,
  DependenciesTabContent,
} from '@/components/explorer-grid/selected-item/tabs-dependencies/index.tsx'

vi.mock(
  '@/components/explorer-grid/selected-item/context.tsx',
  () => ({
    useSelectedItemStore: vi.fn(),
    SelectedItemProvider: 'gui-selected-item-provider',
  }),
)

vi.mock('@/components/ui/tabs.tsx', () => ({
  Tabs: 'gui-tabs',
  TabsList: 'gui-tabs-list',
  TabsTrigger: 'gui-tabs-trigger',
  TabsContent: 'gui-tabs-content',
}))

vi.mock('lucide-react', () => ({
  Blocks: 'gui-blocks-icon',
}))

vi.mock('@/components/ui/data-badge.tsx', () => ({
  DataBadge: 'gui-data-badge',
}))

vi.mock(
  '@/components/explorer-grid/selected-item/tabs-dependencies/empty-state.tsx',
  () => ({
    EmptyState: 'gui-empty-state',
  }),
)

vi.mock(
  '@/components/explorer-grid/selected-item/tabs-dependencies/tabs-insights.tsx',
  () => ({
    InsightsTabContent: 'gui-insights-tab-content',
    InsightsTabButton: 'gui-insights-tab-button',
  }),
)
vi.mock(
  '@/components/explorer-grid/selected-item/tabs-dependencies/tabs-licenses.tsx',
  () => ({
    LicensesTabContent: 'gui-licenses-tab-content',
    LicensesTabButton: 'gui-licenses-tab-button',
  }),
)
vi.mock(
  '@/components/explorer-grid/selected-item/tabs-dependencies/tabs-funding.tsx',
  () => ({
    FundingTabContent: 'gui-funding-tab-content',
    FundingTabButton: 'gui-funding-tab-button',
  }),
)
vi.mock(
  '@/components/explorer-grid/selected-item/tabs-dependencies/tabs-duplicates.tsx',
  () => ({
    DuplicatesTabContent: 'gui-duplicates-tab-content',
    DuplicatesTabButton: 'gui-duplicates-tab-button',
  }),
)

expect.addSnapshotSerializer({
  serialize: v => html(v),
  test: () => true,
})

afterEach(() => {
  const CleanUp = () => (useStore(state => state.reset)(), '')
  render(<CleanUp />)
  cleanup()
})

test('DependenciesTabsButton renders correctly', () => {
  vi.mocked(useSelectedItemStore).mockImplementation(selector =>
    selector({
      selectedItem: SELECTED_ITEM,
      activeTab: 'dependencies',
      setActiveTab: vi.fn(),
      manifest: null,
      rawManifest: null,
      packageScore: undefined,
      insights: undefined,
      author: undefined,
      favicon: undefined,
      publisher: undefined,
      publisherAvatar: undefined,
      versions: undefined,
      greaterVersions: undefined,
      depCount: 2,
      setDepCount: vi.fn(),
      scannedDeps: undefined,
      setScannedDeps: vi.fn(),
      depsAverageScore: undefined,
      setDepsAverageScore: vi.fn(),
      depLicenses: undefined,
      setDepLicenses: vi.fn(),
      depWarnings: undefined,
      setDepWarnings: vi.fn(),
      duplicatedDeps: undefined,
      setDuplicatedDeps: vi.fn(),
      depFunding: undefined,
      setDepFunding: vi.fn(),
    }),
  )

  const Container = () => {
    return <DependenciesTabsButton />
  }
  const { container } = render(<Container />)
  expect(container.innerHTML).toMatchSnapshot()
})

test('DependenciesTabContent renders default', () => {
  vi.mocked(useSelectedItemStore).mockImplementation(selector =>
    selector({
      selectedItem: SELECTED_ITEM,
      activeTab: 'dependencies',
      setActiveTab: vi.fn(),
      manifest: null,
      rawManifest: null,
      packageScore: undefined,
      insights: undefined,
      author: undefined,
      favicon: undefined,
      publisher: undefined,
      publisherAvatar: undefined,
      versions: undefined,
      greaterVersions: undefined,
      depCount: 2,
      setDepCount: vi.fn(),
      scannedDeps: undefined,
      setScannedDeps: vi.fn(),
      depsAverageScore: undefined,
      setDepsAverageScore: vi.fn(),
      depLicenses: undefined,
      setDepLicenses: vi.fn(),
      depWarnings: undefined,
      setDepWarnings: vi.fn(),
      duplicatedDeps: undefined,
      setDuplicatedDeps: vi.fn(),
      depFunding: undefined,
      setDepFunding: vi.fn(),
    }),
  )

  const Container = () => {
    return <DependenciesTabContent />
  }
  const { container } = render(<Container />)
  expect(container.innerHTML).toMatchSnapshot()
})

test('DependenciesTabContent renders an empty state', () => {
  vi.mocked(useSelectedItemStore).mockImplementation(selector =>
    selector({
      selectedItem: SELECTED_ITEM,
      activeTab: 'dependencies',
      setActiveTab: vi.fn(),
      manifest: null,
      rawManifest: null,
      packageScore: undefined,
      insights: undefined,
      author: undefined,
      favicon: undefined,
      publisher: undefined,
      publisherAvatar: undefined,
      versions: undefined,
      greaterVersions: undefined,
      depCount: 2,
      setDepCount: vi.fn(),
      scannedDeps: undefined,
      setScannedDeps: vi.fn(),
      depsAverageScore: undefined,
      setDepsAverageScore: vi.fn(),
      depLicenses: undefined,
      setDepLicenses: vi.fn(),
      depWarnings: undefined,
      setDepWarnings: vi.fn(),
      duplicatedDeps: undefined,
      setDuplicatedDeps: vi.fn(),
      depFunding: undefined,
      setDepFunding: vi.fn(),
    }),
  )

  const Container = () => {
    return <DependenciesTabContent />
  }
  const { container } = render(<Container />)
  expect(container.innerHTML).toMatchSnapshot()
})
