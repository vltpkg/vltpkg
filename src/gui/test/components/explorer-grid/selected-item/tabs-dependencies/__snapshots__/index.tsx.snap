// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DependenciesTabContent renders an empty state 1`] = `

<gui-tabs-content value="dependencies">
  <gui-tabs value="insights">
    <gui-tabs-list variant="nestedCard">
      <gui-insights-tab-button>
      </gui-insights-tab-button>
      <gui-licenses-tab-button>
      </gui-licenses-tab-button>
      <gui-duplicates-tab-button>
      </gui-duplicates-tab-button>
      <gui-funding-tab-button>
      </gui-funding-tab-button>
    </gui-tabs-list>
    <gui-router-outlet>
    </gui-router-outlet>
  </gui-tabs>
</gui-tabs-content>

`;

exports[`DependenciesTabContent renders default 1`] = `

<gui-tabs-content value="dependencies">
  <gui-tabs value="insights">
    <gui-tabs-list variant="nestedCard">
      <gui-insights-tab-button>
      </gui-insights-tab-button>
      <gui-licenses-tab-button>
      </gui-licenses-tab-button>
      <gui-duplicates-tab-button>
      </gui-duplicates-tab-button>
      <gui-funding-tab-button>
      </gui-funding-tab-button>
    </gui-tabs-list>
    <gui-router-outlet>
    </gui-router-outlet>
  </gui-tabs>
</gui-tabs-content>

`;

exports[`DependenciesTabsButton renders correctly 1`] = `

<gui-tabs-trigger
  variant="ghost"
  value="dependencies"
  classname="w-fit px-2"
>
  Dependencies
  <gui-data-badge
    variant="count"
    classnames="[object Object]"
    content="2"
  >
  </gui-data-badge>
</gui-tabs-trigger>

`;
